blaze-bs4/toast/toast.js

```diff
@@ -0,0 +1,104 @@
+import { Template } from 'meteor/templating'
+import { Random } from 'meteor/random';
+import { BlazeBs4 } from 'meteor/jkuester:blaze-bs4';
+
+import './toast.html';
+
+BlazeBs4.toast.add = function({ label = "test", body = "Toast Body", autohide = true, delay = 5000, headerBg = "warning", headerFg = "light", transparent = true, small = '', showId = false, width = 'inherit', animation = true, prepend = true }) {
+  const self = this;
+  var id = Random.id();
+  if (showId) small = id;
+  const toast = {
+    id: id,
+    active: true,
+    rendered: false,
+    label: label,
+    bg: headerBg,
+    fg: headerFg,
+    autohide: autohide,
+    delay: delay,
+    body: body,
+    small: small,
+    transparent: transparent,
+    width: width,
+    animation: animation,
+  };
+  if (self.parentNode) {
+    const nextNode = document.getElementsByClassName('toast');
+    var view = null;
+    if(nextNode.length && prepend === true){
+      view = Blaze.renderWithData(Template.toast_entry, toast, self.parentNode, nextNode[0]);
+    }else{
+      view = Blaze.renderWithData(Template.toast_entry, toast, self.parentNode);
+    }
+    if (BlazeBs4.toast.debug) console.log(`created toast ${toast.id}`, view);
+  } else {
+    console.error(`BlazeBs4.toast: no parentNode`, self);
+  }
+};
+
+Template.toast.onRendered(function () {
+  const instance = this;
+  instance.toasts = instance.find('div.toasts');
+  // Toast.parentNode = instance.find('#toasts');
+  BlazeBs4.toast.parentNode = instance.find('#toasts');
+  if (instance.data) {
+    if (instance.data.width != undefined) {
+      // console.log(`set width to ${instance.data.width}`);
+      instance.toasts.style.width = instance.data.width;
+      instance.toasts.style.maxWidth = instance.data.width;
+    }
+  }
+  //
+  // keep toasts visible when scrolling
+  //
+  var $document = $(document),
+    className = 'hasScrolled';
+
+  $document.scroll(function () {
+    const s = $document.scrollTop();
+    if (s >= document.querySelector('nav').offsetHeight) {
+      instance.toasts.classList.add(className);
+    } else {
+      instance.toasts.classList.remove(className);
+    }
+  });
+});
+
```

Comment:
Nice!

Reply:
Please no `var`, always use `const` and if the variable needs to be re-assigned, use `let`:

https://eslint.org/docs/rules/no-unused-vars